src/software/time/time.cpp

```diff
@@ -23,3 +20,27 @@ double Time::getMilliseconds() const
 }
 
 Time::~Time() {}
+
+double Time::now()
+{
+    const auto clock_now = std::chrono::system_clock::now();
+    return static_cast<double>(std::chrono::duration_cast<std::chrono::microseconds>(
+                                   clock_now.time_since_epoch())
+                                   .count()) /
+           1000000.0;
+}
+
+double Time::secondsSince(std::chrono::time_point<std::chrono::system_clock> start_time)
```

Comment:
could we change this to `Duration Time::timeSince(Time start_time)` instead? Using the std time stuff like this somewhat breaks the abstraction this class is trying to make.

Reply:
Another option is to implement `operator-()` for `Time`, which seems a tad more ambiguous, but isn't too bad IMO

Reply:
this is used for tests, probably better off in test utils